h2/src/main/org/h2/value/Transfer.java

```diff
@@ -471,6 +471,16 @@ public void writeValue(Value v) throws IOException {
             }
             break;
         }
+        case Value.ROW: {
+            ValueRow va = (ValueRow) v;
+            Value[] list = va.getList();
+            int len = list.length;
+            writeInt(len);
+            for (Value value : list) {
+                writeValue(value);
+            }
+            break;
+        }
```

Comment:
this is going to need a protocol version bump and some kind of backwards-compat conversion for clients that are older than the server

Reply:
You mean that older client may ask sever for something like `SELECT (1, 1)`? Yes, such situation is possible, I missed that.

Reply:
Done for both row value and interval data types.

However, there is a problem with old versions. 1.4.194 can read them properly, but 1.4.192 has a problem with metadata, if it does not know reported column type it cannot return values with `ResultSet.getObject()`. (I didn't test 1.4.193).

Reply:
I think this is good enough, there is only so much we can do if people use new features in combination with old client versions